icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -51,15 +54,86 @@ void UnitsTest::runIndexedTest(int32_t index, UBool exec, const char *&name, cha
 
 // Just for testing quick conversion ability.
 double testConvert(UnicodeString source, UnicodeString target, double input) {
-    if (source == u"meter" && target == u"foot" && input == 1.0)
-        return 3.28084;
+    if (source == u"meter" && target == u"foot" && input == 1.0) return 3.28084;
 
-    if ( source == u"kilometer" && target == u"foot" && input == 1.0)
-        return 328.084;
+    if (source == u"kilometer" && target == u"foot" && input == 1.0) return 328.084;
 
     return -1;
 }
 
+void UnitsTest::testConversionCapability() {
+    struct TestCase {
+        const StringPiece source;
+        const StringPiece target;
+        const UnitsConvertibilityState expectedState;
+    } testCases[]{
+        {"meter", "foot", CONVERTIBLE},                                                    //
+        {"kilometer", "foot", CONVERTIBLE},                                                //
+        {"hectare", "square-foot", CONVERTIBLE},                                           //
+        {"kilometer-per-second", "second-per-meter", RECIPROCAL},                          //
+        {"square-meter", "square-foot", CONVERTIBLE},                                      //
+        {"kilometer-per-second", "foot-per-second", CONVERTIBLE},                          //
+        {"square-hectare", "p4-foot", CONVERTIBLE},                                        //
+        {"square-kilometer-per-second", "second-per-square-meter", RECIPROCAL},            //
+        {"g-force", "meter-per-square-second", CONVERTIBLE},                               //
```

Comment:
FYI: from here to line 117 (volt) were a copy of the failing tests from unitsTest.txt - I think they can be deleted again once unitsTest.txt has been connected. Is it worth a TODO to remind us?

Reply:
Great Idea.